### PR TITLE
Region file (.mca) handling.

### PR DESCRIPTION
**Pull request by toqueteos**
**DO NOT MERGE YET.**

First step towards having a chunk loader.
- [x] `.mca` file reading.
- [x] `.mca` file writing.
- [ ] Chunk loader capabilities.

### CODE REVIEW

File: src/vanilla/mod.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+use std::env;
+use std::path::PathBuf;
+
+fn var(key: &str) -> String {
+    match env::var(key) {
+        Ok(val) => val,
+        Err(err) => panic!("couldn't find env var {}; err: {:?}", key, err)
```
Comment (toqueteos) [inline src/vanilla/mod.rs:7]: Is `panic!` a good idea for this?

Comment (fenhl) [inline src/vanilla/mod.rs:7]: No, an envar being unset is very much possible. This function should be removed and calls to it replaced with calls to `env::var` and proper error handling.

Comment (toqueteos) [inline src/vanilla/mod.rs:7]: When are `$HOME` or `%APPDATA%` not set? That's weird!
Also, moving the error handling to `root_path` means changing signature.. `io::Result` or `Option`?

**Comment by toqueteos** [comment]
File writing is utterly incomplete.

File: src/vanilla/mod.rs (hunk 1 above)
Comment (fenhl) [inline src/vanilla/mod.rs:7]: I'd say `io::Result`, but then again I don't know for what purpose we need `root_path` in the first place.

Comment (toqueteos) [inline src/vanilla/mod.rs:7]: Easy vanilla world loading/serving capabilities.

Comment (fenhl) [inline src/vanilla/mod.rs:7]: Minecraft servers are usually not run from the .minecraft directory.

**Comment by atheriel** [comment]
@toqueteos I'm working on `#[derive]` functionality for NBT file formats that should make this task massively easier.

**Comment by toqueteos** [comment]
Hmmm... Is `#[derive]` customizable now? That's awesome!

**Comment by fenhl** [comment]
`#[derive(NbtFmt)]` has landed now.

**Comment by toqueteos** [comment]
I'm closing this PR in favor of a new updated one.

**Pull request closed without merging**